mempool/docs/config.md

```diff
@@ -0,0 +1,59 @@
+# Mempool Configuration
```

Comment:
there is https://github.com/tendermint/tendermint/pull/792. let's try to avoid duplication. thanks

Reply:
@zramsay maybe you can incorporate some things below into your PR